### PR TITLE
Add filewriter script

### PR DESCRIPTION
**Pull request by repasics**

### CODE REVIEW

**Review by loki04 — CHANGES_REQUESTED** [review]
I suggest to add this into a `utils.py` instead like in #2 !
And remove the usages from other places as well

**Comment by repasics** [comment]
I Added here the settings.py. I will remove it from #3

File: src/scripts/utils.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    while True:
+        filename = settings._GENERATED_FILENAME.format(NUMBER = file_count)
+        file_to_open = os.path.join(output_dir, filename)
+        with open(file_to_open, "w") as file:
+            file.write("%s\n" % (settings._FILE_HEADER))
+            counter = settings._TEST_CASE_IN_A_FILE
+            for i in range(len(content)):
+                file.write("%s" %(content[0]))
+                del content[0]
+                counter -= 1
```
Comment (rerobika) [inline src/scripts/utils.py]: What is the reason of removing elements from the list? Cannot we just simply iterate through the list elements?

File: src/scripts/settings.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+from os import path
+
+SCRIPTS_DIR = path.dirname(path.abspath(__file__))
+TEMPLATE_DIR = path.normpath(path.join(SCRIPTS_DIR, 'templates'))
+PROJECT_DIR = path.normpath(path.join(SCRIPTS_DIR, '..', '..'))
+NUMBER_DIR = path.normpath(path.join(PROJECT_DIR, 'src', 'number'))
+
+class file_write_settings():
+    def __init__(self, generated_filename, output_dir, file_header, test_case_in_a_file):
+        self._GENERATED_FILENAME = generated_filename # The template of the generated filename
```
Comment (rerobika) [inline src/scripts/settings.py]: `_GENERATED_FILENAME` make it lowercase. (same as for each element below)

**Review by rerobika — CHANGES_REQUESTED** [review]

**Comment by repasics** [comment]
Done the requested changes

File: src/scripts/settings.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+TEMPLATE_DIR = path.normpath(path.join(SCRIPTS_DIR, 'templates'))
+PROJECT_DIR = path.normpath(path.join(SCRIPTS_DIR, '..', '..'))
+NUMBER_DIR = path.normpath(path.join(PROJECT_DIR, 'src', 'number'))
+
+class file_write_settings():
+    def __init__(self, generated_filename, output_dir, file_header, test_case_in_a_file):
+        self.generated_filename = generated_filename # The template of the generated filename
+        self.output_dir = output_dir # The output directory
+        self.file_header = file_header # The header of the test files
+        self.test_case_in_a_file = test_case_in_a_file # The number of test cases for each file
```
Comment (rerobika) [inline src/scripts/settings.py]: `test_case_in_a_file`  -> `test_cases_in_a_file`

File: src/scripts/utils.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    test_case_count = 0
+    if not os.path.isdir(output_dir):
+        os.makedirs(output_dir)
+    while True:
+        filename = settings.generated_filename.format(NUMBER = file_count)
+        file_to_open = os.path.join(output_dir, filename)
+        with open(file_to_open, "w") as file:
+            file.write("%s\n" % (settings.file_header))
+            counter = settings.test_case_in_a_file
+            for i in range(test_case_count, len(content)):
```
Comment (rerobika) [inline src/scripts/utils.py:16]: `for in` can be used in this case

File: src/scripts/utils.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+                file.write("%s" %(content[i]))
+                counter -= 1
+                test_case_count += 1
```
Comment (rerobika) [inline src/scripts/utils.py:18]: `test_case_count` is increases while `counter` decreases in each iteration. How about merging these values?

**Review by rerobika — CHANGES_REQUESTED** [review]

File: src/scripts/utils.py (hunk 4 above)
Comment (repasics) [reply src/scripts/utils.py:16]: I think this kind of for loop is the best choice for this

File: src/scripts/utils.py (hunk 5 above)
Comment (repasics) [reply src/scripts/utils.py:18]: Done

File: src/scripts/utils.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python
+import os
+from constants import _constants
+
+def _write_file(settings, content):
```
Comment (rerobika) [inline src/scripts/utils.py]: The _ prefix is not needed, since it's an util function.

**Review by rerobika — CHANGES_REQUESTED** [review]

File: src/scripts/utils.py (hunk 6 above)
Comment (repasics) [reply src/scripts/utils.py]: I removed it

**Review by rerobika — APPROVED** [review]

**Review by loki04 — APPROVED** [review]
LGTM

**Pull request merged**